state.go

```diff
@@ -930,8 +930,14 @@ func (m *Memberlist) aliveNode(a *alive, notify chan struct{}, bootstrap bool) {
 	} else {
 		// Check if this address is different than the existing node unless the old node is dead.
 		if !bytes.Equal([]byte(state.Addr), a.Addr) || state.Port != a.Port {
+			// If DeadNodeReclaimTime is configured, check if enough time has elapsed since the node died.
+			var canReclaim bool
+			if m.config.DeadNodeReclaimTime > 0 && state.StateChange.Add(m.config.DeadNodeReclaimTime).Before(time.Now()) {
```

Comment:
nitpick: I think this is correct but I'd find it slightly more intuitive to write this as:

```go
canReclaim := (m.config.DeadNodeReclaimTime > 0 && 
  time.Since(state.StateChange) > m.config.DeadNodeReclaimTime)
```